### ISSUE TITLE
Add flake8 check

### ISSUE THREAD

**Issue by refeed** [kiwix/kiwix-build#263]
`kiwix-build` mainly comprises of Python codes, I think it'd be nice if this repository had a flake8 (which checks for codestyle and possible errors and bug (such as unused var, etc)) check in its Travis-CI.

**Comment by kelson42** [comment]
@refeed Why not, would be be able to make a PR to introduce it?

**Comment by refeed** [comment]
@refeed Alright, I'm gonna make a PR for it

**Comment by HimanshuMahto** [comment]
is it still open if then i would like to work on it.

**Comment by kelson42** [comment]
@HimanshuMahto yes you can

**Comment by Rayan-Rasheed** [comment]
Is this issue still open? Because I want to work on it.

**Comment by kelson42** [comment]
@mgautierfr ?

**Comment by mgautierfr** [comment]
The issue is still open. But I strongly suggest that @Rayan-Rasheed and @HimanshuMahto discuss together before doing both the same work.
On my side I don't care about who does it. I will review the first PR coming (and accept the first acceptable one). I don't which to review to almost identical PR.

**Comment by Vishnurr2k01** [comment]
is this issue still available ?

**Comment by mgautierfr** [comment]
Yes.

**Comment by Anandbrewery** [comment]
is this issue still open?
i am interested to contribute for this

**Comment by kelson42** [comment]
Yes, I still want to have a formalism check of the Python code... but we should probably revide a bit the requirements. @benoit74 @rgaudin Could you please tell what should be respected to respect our standards? Maybe you could also point to a place where the check workflow ismproperly implemented already?

**Comment by rgaudin** [comment]
https://github.com/openzim/_python-bootstrap is our bootstrap repo for Python project. The workflows are explicit I believe and rely on a couple things:

- replacing the `setup.py` and `MANIFEST.in` with a `pyproject.toml` file that both describes the project and configures all tools
- a `tasks.py` file that contains the logic for those always-similar checks.

Once in place, we have 3 main checks:

- `lint:all` that checks linting using black and ruff`.
- `check:all` that runs the type checker
- `test:coverage` that runs the tests and outputs the coverage as well.

The first one is the only one discussed and wanted here. I think `check` might be wanted later but it's an effort to fix and requires good knowledge of the codebase to not break anything. There are no tests as you know so the tests and coverage parts are useless as well.

I would thus suggest to keep the  bootstrap config, copy the whole tasks.py but tweak the workflow to only run linting.

It's not trivial but it shouldn't be difficult.

**Comment by kelson42** [comment]
@rgaudin Thank you!
@Anandbrewery Still motivated?

**Comment by Anandbrewery** [comment]
Yes I'm and Happy new year 2025!

if i am stuck whom should i reach out @kelson42 @rgaudin

**Comment by rgaudin** [comment]
Myself
